### DISCUSSION TITLE
set_interval possible somehow?

### DISCUSSION THREAD

**Discussion by slauko** [idom-team/idom#1132] [Question] [answered]
hey, 
i just found this nice project today and was experimenting with it a bit, 
im just wondering if there is any set_interval api i could use to like constantly update data in a use_effect hook?

any advice appreciated :)

greetings

**Comment by Archmonger** [comment, marked as answer]
The most common way of doing this is via a `while True` loop within your `use_effect` hook.

```python
import asyncio

from reactpy import component, use_effect


@component
def example():
    @use_effect
    async def periodic_task():
        while True:
            await asyncio.sleep(1)
            do_something()
```

**Reply by slauko** [reply]
ah okay, 
thanks for the fast answer, 
i was about to handle it like this with threading, 
is there any downside or benefit using threading or asyncio?

```py
    def set_interval(func, sec):
        def func_wrapper():
            set_interval(func, sec)
            func()
        t = threading.Timer(sec, func_wrapper)
        t.start()
        return t
   
    def clear_interval(timer):
        timer.cancel()
    
    @component
    def App():
        counter, set_counter = use_state(0)

        def update_counter():
            data = api.get_data()
            counter = data.get("counter", 0)
            set_counter(counter)

        def _effect():
            timer = set_interval(update_counter, 0.01)
            return lambda: clear_interval(timer)

        use_effect(_effect, None)

**Reply by Archmonger** [reply]
Asyncio approach is easier to read, and additionally has less performance overhead than creating a new thread for every single app component.

**Reply by slauko** [reply]
ok, thanks, just noticed that the app gets slower pretty fast after a cuple of iterations with threads 😅

**Reply by rmorshea** [reply]
If the problem is a proliferation of threads you could use a thread pool. Async more generally avoids this problem though.
